### PR TITLE
Allow deploys to the same branch name in a different repo

### PR DESCRIPTION
**Pull request by sigmachirality**
This duplicates https://github.com/enriikke/gatsby-gh-pages-action/pull/28 and https://github.com/enriikke/gatsby-gh-pages-action/pull/23 but I'm opening another PR so I can make a case for why I think this change is necessary/explain our use case.

Currently we have a repo for our [code](https://github.com/CIS192/website) and [another that is configured to work](https://github.com/CIS192/cis192.github.io) with Github Pages. I wanted to use this action to setup a workflow that triggers whenever someone commits to master and immediately redeploys the site. However, since I want to trigger builds from website/master to cis192.github.io/master, the action did not allow this build because it detected that the branches had the same name (master).

I understand that this check is necessary to prevent recursive deploys (a deploy triggering itself). However, I there isn't a danger of this happening when the branches are different repos (unless you for some reason decide to setup another workflow that deploys back to the source code repo 🤷). Therefore I think it makes sense to make this change to support our use case, especially since I don't think it would make sense to trigger builds or deploy on branches other than master in the two repos we're working with.

I am very new to anything dev-ops/GH actions related, so if @enriikke has reasons behind not accepting this change, I'd love to know. I could be very wrong here and I'd like to know if I am.

I hope this change goes through, but in the meanwhile I've forked the repo and am running a custom release of this action, so to anyone reading this in perpetuity: if you're running into this problem and this PR or the other two I mentioned haven't been merged yet, you might be able to use my fork if I haven't deleted it.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -19,7 +19,10 @@ async function run(): Promise<void> {
     let deployBranch = core.getInput('deploy-branch')
     if (!deployBranch) deployBranch = DEFAULT_DEPLOY_BRANCH
 
-    if (github.context.ref === `refs/heads/${deployBranch}`) {
+    const deployRepo = core.getInput('deploy-repo')
+    const isSameRepo = !deployRepo || deployRepo === github.context.repo.repo
+
+    if (isSameRepo && github.context.ref === `refs/heads/${deployBranch}`) {
```
Comment (enriikke) [inline index.ts:25]: Would you mind adding a new test to validate this? 😃 

There is an existing one that checks skipping the build if the branch is the same: https://github.com/enriikke/gatsby-gh-pages-action/blob/84aa969a7918a9488d4dac31f2d847274b31ff07/__tests__/index.test.ts#L72-L77

I'm thinking we need a similar one that actually calls `build` if the branches are the same but the repo is different. I think you could validate the `build` call at the end by doing something like this: https://github.com/enriikke/gatsby-gh-pages-action/blob/84aa969a7918a9488d4dac31f2d847274b31ff07/__tests__/index.test.ts#L84

File: dist/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
...ncoding(r)}u.on("data",e=>{a.push(e);if(s){o=a.length}else{o+=e.length}});u.getBufferedValue=(()=>{if(t){return a}return i?Buffer.concat(a,o):a.join("")});u.getBufferedLength=(()=>o);return u})},969:function(e,t,r){var n=r(11);e.exports=n(once);e.exports.strict=n(onceStrict);once.proto=once(function(){Object.defineProperty(Function.prototype,"once",{value:function(){return once(this)},configurable:true});Object.defineProperty(Function.prototype,"onceStrict",{value:function(){return onceStrict(this)},configurable:true})});function once(e){var t=function(){if(t.called)return t.value;t.called=true;return t.value=e.apply(this,arguments)};t.called=false;return t}function onceStrict(e){var t=function(){if(t.called)throw new Error(t.onceError);t.called=true;return t.value=e.apply(this,arguments)};var r=e.name||"Function wrapped with `once`";t.onceError=r+" shouldn't be called more than once";t.called=false;return t}},986:function(e,t,r){"use strict";var n=this&&this.__awaiter||function(e,t,r,n){function adopt(e){return e instanceof r?e:new r(function(t){t(e)})}return new(r||(r=Promise))(function(r,i){function fulfilled(e){try{step(n.next(e))}catch(e){i(e)}}function rejected(e){try{step(n[
```
Comment (enriikke) [inline dist/index.js:1]: Could you also remove this file? 😃

**Review by enriikke — COMMENTED** [review]
👋 hi @sigmachirality! Thanks for taking the time to provide context behind the need of this feature. That makes a lot of sense to me and I'll be happy to merge your PR. 

I left a couple of comments. Would you mind addressing those so we can get this merged? 😃

**Comment by sigmachirality** [comment]
uh oh

I'm having some issues getting the test to work properly, I copied the syntax of the other tests but it runs yarn install instead of yarn build. I have a limited understanding of how GH-Actions are tested, so @enriikke can you take a look and lmk how to fix the test or fix the test yourself?

**Comment by sigmachirality** [comment]
why

why2

Seeing as there was no response, I investigated a bit further on my own time. My experience with jest is limited, but I discovered the above behavior: when called without setting 'gatsby-args' specifically in the input, the current method of structuring test cases creates a test that fails. As far as I can tell, this is not intended test behavior, as 'gatsby-args' should default to '', making it no different than setting input['gatsby-args'] = ''. If 'calls gatsby build with args' passes, this case should pass as well.

As an interim measure, I've set 'gatsby-args' to '' in the new test case that @enriikke requested above. After setting gatsby-args, this new test case passes. The action with the proposed change in this PR is working fine for our website/repo setup, so I conclude that there is a good chance the current tests are fragile/not testing properly, but I lack context and understanding of the current code to definitively conclude this or solve such a problem if it exists. I recommend @enriikke to investigate this further.

File: index.ts (hunk 1 above)
Comment (sigmachirality) [reply index.ts:25]: New test case is added, see new comments for further info.

**Comment by enriikke** [comment]
👋 hi @sigmachirality! sorry for the late reply, it's been a bit of a busy month for me. I'll take a look at this over the weekend. Thanks for digging further! 😃

**Comment by enriikke** [comment]
Hi @sigmachirality! Thanks for adding this into the Action. I did some digging and will investigate the test setup further in a separate branch. 😃

**Comment by sigmachirality** [comment]
Alright, sounds good!

**Pull request merged**